server/plugin/plugin.go

```diff
@@ -178,6 +180,14 @@ func (p *Plugin) OnConfigurationChange() (err error) {
 		return errors.New("status sync job interval should be greater than 0")
 	}
 
+	if err = command.UnRegister(pluginAPIClient); err != nil {
+		return errors.Wrap(err, "failed to unregister slash command")
+	}
+
+	if err = command.Register(pluginAPIClient, stored.AutoConnectUsers); err != nil {
+		return errors.Wrap(err, "failed to register slash command")
+	}
```

Comment:
These two blocks should be executed only if `previousConfig.AutoConnectUsers != config.AutoConnectUsers`.